libindy/src/services/route/mod.rs

```diff
@@ -0,0 +1,523 @@
+use domain::route::*;
+use domain::crypto::key::Key;
+use domain::crypto::combo_box::ComboBox;
+use errors::route::RouteError;
+use errors::common::CommonError;
+use services::crypto::CryptoService;
+use services::wallet::{WalletService, RecordOptions};
+use utils::crypto::base64::{decode, encode};
+use utils::crypto::xsalsa20::{encrypt_payload, decrypt_payload};
+use utils::serialization::jwm::*;
+use std::rc::Rc;
+
+pub struct RouteService { }
+
+impl RouteService {
+    pub fn new() -> RouteService {
+        RouteService {}
+    }
+
+    pub fn pack_msg(&self, plaintext: &str, recv_keys: &Vec<String>, my_vk: Option<&str>, is_authcrypt: bool,
```

Comment:
This approach definitely works (awesome job, btw, really looking forward to having this) but I think that separating out auth and anon pack out into separate functions may improve overall understandability of this method. Is there a compelling reason to mix both into the one function?

Reply:
when I originally designed it I was combining them. With this new iteration I've been rewriting they are now separate making for much cleaner code to read.